compiler/ras/Debug.cpp

```diff
@@ -596,11 +596,10 @@ TR_Debug::vtrace(const char * format, va_list args)
    if (_file != NULL)
       {
       char buffer[256];
-      if(TR::Options::_traceFileLength && TR::IO::ftell(_file) > (TR::Options::_traceFileLength * (1 << 20)))
+      if ((0 != TR::Options::_traceFileLength) && (TR::IO::ftell(_file) > (((long)TR::Options::_traceFileLength) << 20)))
```

Comment:
If we're cleaning this up anyway then it would be better to:
1. Split the conditions into two lines for better readability.
2. Use `static_cast<uint64_t>(TR::Options::_traceFileLength)` since `long` can be 32 bits even on 64 bit platforms.

Technically there's even an argument to be mad that the trace file length shouldn't be a signed integer to begin with since it's supposed to be a non-negative number.

Reply:
Ah I see, I think then probably have to do something like
```
static_cast<int64_t>(TR::IO::ftell(_file)) > static_cast<int64_t>(TR::Options::_traceFileLength << 20)
```
because as it stands `1 << 20` is more than 32 bits; I don't know enough about the C++ spec to know if the LHS and RHS gets promoted to 64 bits prior to the comparison or not.

Reply:
That approach seems reasonable, but the order of operations is wrong: the cast needs to occur before the left-shift. I'll update my branch accordingly.

Reply:
Updated.